LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/LogSink.cs

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan
+{
+    using System;
+    using System.Buffers;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Channels;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Primitives;
+
+    public interface ILogSink
+    {
+        public LogLevel LogLevel { get; }
+        public void Log(LogLevel logLevel, string message);
+    }
+
+    public class LogSink : ILogSink
+    {
+        public LogSink(LogLevel logLevel) => LogLevel = logLevel;
+
+        public LogLevel LogLevel { get; }
+
+        public virtual void Log(LogLevel logLevel, string message)
+        {
+            if (LogLevel > logLevel)
+                return;
+            CoreLog(logLevel, message);
+        }
+
+        protected virtual void CoreLog(LogLevel logLevel, string message) { /* Nop */ }
```

Comment:
shouldn't this be an abstract?

Reply:
The idea was to allow an instance of `LogSink` to be used as a null/NOP logger, which I had originally but went away with 1cee7f002352918ba5efc70998869fb8cd25d322. I can make it abstract now.

Reply:
Made abstract with dd3c84fc26e6103f8684a392a4315de6dd0b9588.